internal/mode/static/nginx/config/base_http_config_template.go

```diff
@@ -0,0 +1,5 @@
+package config
+
+const baseHTTPTemplateText = `
+{{- if .HTTP2 }}http2 on;{{- end }}
```

Comment:
```suggestion
{{- if .HTTP2 }}http2 on;{{ end }}
```

I think we only need the single hyphen just so to preserve the following newline. Keeps some spacing between this directive and the next (which might be server).